lib/commons/color/get-background-color.js

```diff
@@ -57,11 +94,81 @@ function calculateObscuringAlpha(elmIndex, elmStack) {
 	}
 	return totalAlpha;
 }
+/**
+ * Determine if element is partially overlapped, triggering a Can't Tell result
+ * @private
+ * @param {Element} elm
+ * @param {Element} bgElm
+ * @param {Object} bgColor
+ * @return {Boolean}
+ */
+function elmPartiallyObscured(elm, bgElm, bgColor) {
+	var obscured = (elm !== bgElm && !dom.visuallyContains(elm, bgElm) && bgColor.alpha !== 0);
+	if (obscured) {
+		axe.commons.color.incompleteData.set('bgColor', 'elmPartiallyObscured');
+	}
+	return obscured;
+}
 
-function elmOutsideParent(elm, bgElm, bgColor) {
-	var visible = (elm !== bgElm && !dom.visuallyContains(elm, bgElm) && bgColor.alpha !== 0);
-	return visible;
+/**
+ * Include nodes missing from initial gathering because
+ * document.elementsFromPoint misses some elements we need
+ * i.e. TR is missing from table elementStack and leaves out bgColor
+ * https://github.com/dequelabs/axe-core/issues/273
+ *
+ * @private
+ * @param {Array} elmStack
+ * @param {Element} elm
+ */
+function includeMissingElements(elmStack, elm) {
+	const elementMap = {'TD': 'TR', 'INPUT': 'LABEL'};
+	const tagArray = elmStack.map((elm) => {
+		return elm.tagName;
+	});
+	for (let candidate in elementMap) {
+		if (elementMap.hasOwnProperty(candidate)) {
+			// tagName matches key
+			if (elm.tagName === candidate) {
+				let ancestorMatch = axe.commons.dom.findUp(elm, elementMap[candidate]);
+				if (ancestorMatch && elmStack.indexOf(ancestorMatch) === -1) {
+					// found an ancestor not in elmStack, and it overlaps
+					let overlaps = axe.commons.dom.visuallyOverlaps(elm.getBoundingClientRect(), ancestorMatch);
+					if (overlaps) {
+						elmStack.splice(elmStack.indexOf(elm) + 1, 0, ancestorMatch);
+					}
+				}
+			}
+			// tagName matches value
+			// (such as LABEL, when matching itself. It should be in the list, but Phantom skips it)
+			if (elm.tagName === elementMap[candidate] && tagArray.indexOf(elm.tagName) === -1) {
+				elmStack.splice(tagArray.indexOf(candidate) + 1, 0, elm);
+			}
+		}
+	}
 }
+
+/**
+ * Look at document and body elements for relevant background information
+ * @private
+ * @param {Array} elmStack
+ */
+function consultDocumentBody(elmStack) {
```

Comment:
This could do with a better name, something like `sortPageBackground`. Also, as before, I think this should be a pure function. Don't mutate elmStack, just return a new array. Much easier to test and less error prone.